packages/frontend/src/components/ProjectAccess/ProjectAccessCreation/index.tsx

```diff
@@ -1,242 +1,178 @@
-        isError,
-        isSuccess,
-        isLoading,
-    } = useCreateProjectAccessMutation(projectUuid);
+    const { data: organizationUsers } = useOrganizationUsers();
+    const { mutateAsync: createMutation, isLoading } =
+        useCreateProjectAccessMutation(projectUuid);
 
     const {
-        data: inviteData,
-        mutate: inviteMutation,
+        mutateAsync: inviteMutation,
         isLoading: isInvitationLoading,
-        isSuccess: isInvitationSuccess,
         reset,
     } = useCreateInviteLinkMutation();
 
-    const methods = useForm<CreateProjectMember>({
-        mode: 'onSubmit',
-        defaultValues: {
+    const form = useForm<Pick<CreateProjectMember, 'email' | 'role'>>({
+        initialValues: {
+            email: '',
             role: ProjectMemberRole.VIEWER,
         },
     });
-    const [emailSelected, setEmailSelected] = useState<string>('');
+
     const [addNewMember, setAddNewMember] = useState<boolean>(false);
     const [inviteLink, setInviteLink] = useState<InviteLink | undefined>();
-
-    const { data: organizationUsers } = useOrganizationUsers();
-    const orgUserEmails =
-        organizationUsers && organizationUsers.map((orgUser) => orgUser.email);
+    const [emailOptions, setEmailOptions] = useState<string[]>([]);
 
     useEffect(() => {
-        if (isError) {
-            methods.reset({ ...methods.getValues() }, { keepValues: true });
-        }
-        if (isSuccess) {
-            setEmailSelected('');
-            methods.setValue('role', ProjectMemberRole.VIEWER);
+        if (organizationUsers) {
+            setEmailOptions(organizationUsers.map(({ email }) => email));
         }
-    }, [isError, methods, isSuccess, showToastSuccess, setEmailSelected]);
+    }, [organizationUsers]);
 
-    useEffect(() => {
-        if (isInvitationSuccess) {
-            setInviteLink(inviteData);
-            reset();
-            createMutation({
-                role: methods.getValues('role'),
-                email: emailSelected,
-                sendEmail: false,
-            });
-            setAddNewMember(false);
-        }
-    }, [
-        isInvitationSuccess,
-        inviteData,
-        emailSelected,
-        addNewMember,
-        methods,
-        createMutation,
-        reset,
-    ]);
-
-    const handleSubmit = (formData: CreateProjectMember) => {
+    const handleSubmit = async (
+        formData: Pick<CreateProjectMember, 'email' | 'role'>,
+    ) => {
         track({
             name: EventName.CREATE_PROJECT_ACCESS_BUTTON_CLICKED,
         });
+        setInviteLink(undefined);
 
         if (addNewMember) {
-            inviteMutation({
-                email: emailSelected,
+            const data = await inviteMutation({
+                email: formData.email,
                 role: OrganizationMemberRole.MEMBER,
             });
+            await createMutation({
+                ...formData,
+                sendEmail: false,
```

Comment:
if we add a new member, shouldn't we send an email? 🤔 maybe I'm thinking about this wrong... checking render now as well

Reply:
We send an email when inviting the user to the organization, so we don't send the email inviting him specifically to this project.